src/components/DocLink.tsx

```diff
@@ -1,94 +1,67 @@
-  position: relative;
-  z-index: 1;
-  text-decoration: none;
-  display: flex;
-  flex-direction: row;
-  flex: 1;
-  width: 100%;
-  justify-content: space-between;
-  padding: 1rem;
-  border-radius: 2px;
-  color: ${({ theme }) => theme.colors.text};
-  border: 1px solid ${({ theme }) => theme.colors.border};
-  &:hover {
-    text-decoration: none;
-    box-shadow: 0 0 1px ${({ theme }) => theme.colors.primary};
-    background: ${({ theme }) => theme.colors.tableBackgroundHover};
-    border-radius: 4px;
-
-    svg {
-      fill: ${(props) => props.theme.colors.primary};
-      transition: transform 0.1s;
-      transform: scale(1.2)
-        rotate(${({ isExternal }) => (isExternal ? "-45deg" : "0")});
-    }
-  }
-  @media (max-width: ${({ theme }) => theme.breakpoints.m}) {
-    width: 100%;
-  }
-`
-
-const TextCell = styled.div`
-  flex: 1;
-  flex-direction: column;
-  color: ${({ theme }) => theme.colors.text};
-`
-
-const Title = styled.p`
-  color: ${({ theme }) => theme.colors.text300};
-  font-weight: 600;
-  margin: 0;
-`
-
-const Arrow = styled(Icon)<{ isExternal: boolean }>`
-  margin: 0rem 1.5rem;
-  align-self: center;
-  min-width: 2rem;
-  transform: rotate(${({ isExternal }) => (isExternal ? "-45deg" : "0")});
-`
-
-const EmojiCell = styled.div`
-  display: flex;
-  align-items: center;
-`
-
-const DocLink: React.FC<IProps> = ({
-  to,
-  children,
-  className,
-  isExternal = false,
-}) => {
-  const theme = useTheme()
-
+const DocLink: React.FC<IProps> = ({ to, children, isExternal = false }) => {
   return (
-    <Container to={to} className={className} isExternal={isExternal} hideArrow>
-      <EmojiCell>
-        <Emoji size={1} text=":page_with_curl:" mr={`1rem`} />
-      </EmojiCell>
-      <TextCell>
-        <Title>{children}</Title>
-      </TextCell>
-      <Arrow
-        isExternal={isExternal}
-        color={theme.colors.text}
-        name="arrowRight"
-      />
-    </Container>
+    <LinkBox>
+      <LinkOverlay href={to} isExternal={isExternal}>
+        <Box
+          textDecoration="none"
+          padding={4}
+          borderRadius="sm"
+          color="text"
+          border="1px"
+          borderStyle="solid"
+          borderColor="border"
+          _hover={{
+            textDecoration: "none",
+            background: "tableBackgroundHover",
+            borderRadius: "base",
+          }}
+          role="group"
+        >
+          <Flex
+            position="relative"
+            zIndex={1}
+            display="flex"
+            flexDirection="row"
+            flex={1}
+            justifyContent="space-between"
+          >
+            <Flex align="center">
+              <Emoji fontSize="md" mr={4} text=":page_with_curl:" />
+            </Flex>
+            <Box flex={1} flexDirection="column" color="text">
+              <Text color="text300" fontWeight="semibold" margin={0}>
+                {children}
+              </Text>
```

Comment:
```suggestion
            <LinkOverlay href={to} isExternal={isExternal}>
              <Text color="text300" fontWeight="semibold" margin={0}>
                {children}
              </Text>
            </LinkOverlay>
```

Reply:
We also need to use our custom `Link` in this `LinkOverlay` in order to make it work correctly for internal navigation.

Something like:
```tsx
import Link from "./Link"

<LinkOverlay
  as={Link}
  href={to}
  hideArrow
  color="text"
  textDecoration="none"
  _hover={{ textDecoration: "none" }}
>
```

You can see a similar implementation in this open PR #7721